rfd/0127-encrypted-session-recordings.md

```diff
@@ -0,0 +1,373 @@
+  leaving all other auth servers without a way of retrieving new keys.
+- Replace wrapped private key and active keys marked for rotation in a single
+  atomic write to the backend.
+
+All other auth servers will check if their keys are marked for rotation when a
+change to the resource is picked up by their watcher. If they find their keys
+need to be rotated, they will:
+- Unwrap their copy of the `X25519` private key.
+- Generate a new `RSA` wrapping keypair.
+- Wrap the `X25519` private key.
+- Replace their active key with the newl wrapped key and wrapping keypair in a
+  single atomic write to the backend.
+
+It's worth noting that we do not need to move these keys into the `RotatedKeys`
+list because the expectation is that we will no longer have access to them in
+their respective HSM or KMS. Since the same underlying `X25519` keypair is
+ultimately driving encryption and decryption, we aren't losing any ability to
+decrypt historical sessions.
+
+#### Rotating `Identity` and `Recipient` keys
+
+When an auth server receives a rotation request for the `X25519` keypair, it
+will:
+- Copy all wrapped keys from the list of active keys to the list of rotated
+  keys.
+- Generate a new `X25519` keypair.
+- Iterate over all active keys and use their public `RSA` keys to generate new
+  wrapped private keys.
+- Replace all active and rotated keys in a single atomic write to the backend.
+
+
+#### `tctl` Changes
+
+Key rotation will be handled through `tctl` using a new subcommand:
+```bash
+tctl auth keys rotate --type=recording-data # for rotating X25519 pairs
+tctl auth keys rotate --type=recording-wrap # for rotating HSM/KMS backed wrapping keys
+```
+The reasoning for the new subcommand was to try and avoid cases where you might
+forget the `--type` flag and initiate a rotation of the user and host CAs.
+
+### Security
+
+Protection of key material invovled with encrypting session recordings is
+largely managed by our existing keystore features. The one exception being the
+private keys used by `age` to decrypt files during playback. Whenever possible,
+those keys will be wrapped by the backing keystore such that decryption related
+secrets are never directly accessible.
+
+One of the primary concerns outside of key management is ensuring that session
+recording data is always encrypted before landing on disk or in long term
+storage. In order to help enforce this, all session recording interactions
+should be gated behind a standard interface that can be implemented as either
+plaintext or encrypted. This will help ensure that once the encrypted writer
+has been selected, any interactions with session recordings are encrypted by
+default.
+
+## UX Examples
+
+For the most part, the user experience of encrypted session recordings is
+identical to non-encrypted session recordings. The only notable change is the
+addition of the `tctl auth rotate recordings` subcommand for rotating keys
```

Comment:
Is it `tctl auth rotate recordings` or `tctl auth keys rotate`?

Reply:
It should be `tctl auth keys rotate` for now, this and the example commands should be fixed now :+1: